independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1431,10 +1431,12 @@ void implementCreateForClassBean(ClassOutput classOutput, ClassCreator beanCreat
                     DotNames.POST_CONSTRUCT,
                     bean.getDeployment().getIndex());
             for (MethodInfo callback : postConstructCallbacks) {
-                if (Modifier.isPrivate(callback.flags())) {
-                    privateMembers.add(isApplicationClass,
-                            String.format("@PostConstruct callback %s#%s()", callback.declaringClass().name(),
-                                    callback.name()));
+                if (isReflectionFallbackNeeded(callback, targetPackage)) {
+                    if (Modifier.isPrivate(callback.flags())) {
+                        privateMembers.add(isApplicationClass,
```

Comment:
This only logs private members in the code. Do we want some logging for the pack-private plus package split case too? Because effectively both scenarios force us into reflection which is something we want to avoid, right?

Reply:
I'm not sure. We could log something but the message would have to be quite "elaborate".

Reply:
Yeah and the detection too (since this can happen on multiple places), not sure it's worth it.